02.calendar/cal.rb

```diff
@@ -0,0 +1,64 @@
+#!/usr/bin/env ruby
+
+require 'optparse'
+
+input = ARGV.getopts('y:', 'm:')
+
+now = Time.now
+
+# use now if input not available
+year = input['y'] ? input['y'].to_i : now.year
+month = input['m'] ? input['m'].to_i : now.month
+
+this_month = false
+if now.year == year && now.month == month
+  this_month = true
+end
+
+# Get first day of the month
+start_day = Time.new(year, month).wday
+
+# Get last date of the month
+dates_in_month = (Time.new(year, (month % 12) + 1) - 1).day
```

Comment:
```suggestion
dates_in_month = Date.new(year, month, -1).day
```

Dateクラスを使えば、月の最終日はもう少しシンプルにかけますよ。ご参考までに。